map.ipynb

```diff
@@ -0,0 +1,386 @@
+    "        \n",
+    "    elif not start_date.value and not end_date.value:\n",
+    "        print('Please enter a start and end date before continuing')\n",
+    "      \n",
+    "    else:\n",
+    "        print('Continuing...')\n",
+    "\n",
+    "        \n",
+    "def handle_add(arg):\n",
+    "    if f'{latitude.value}, {longitude.value}' not in points:\n",
+    "        points.append(f'{latitude.value}, {longitude.value}')\n",
+    "        point_list.options = points\n",
+    "    \n",
+    "    new_icon = AwesomeIcon(name=\"map-pin\", marker_color='orange')\n",
+    "    new_marker = Marker(location = (latitude.value, longitude.value), icon=new_icon, draggable=False)\n",
+    "    \n",
+    "    if new_marker not in marker_group.layers:\n",
+    "        clear_marker()\n",
+    "        marker_group.add_layer(new_marker)\n",
+    "    \n",
+    "    \n",
+    "def handle_remove(arg):\n",
+    "    if point_list.value:\n",
+    "        for point in point_list.value:\n",
+    "            points.remove(point)\n",
+    "            \n",
+    "        for marker in marker_group.layers:\n",
+    "            if ', '.join(marker.location) in point_list.value:\n",
+    "                marker_group.remove_layer(marker)\n",
+    "            \n",
+    "    point_list.options = points"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "def clear_marker():\n",
+    "    for i in range(1, len(m.layers)):\n",
+    "        if m.layers[i].name == 'Marker':\n",
+    "            m.remove_layer(m.layers[i])\n",
+    "\n",
+    "\n",
+    "def in_polygon(lat, lon):\n",
+    "\n",
+    "    point = shapely.geometry.Point(lat, lon)\n",
+    "\n",
+    "    peace_polygon = shapely.geometry.polygon.Polygon(p.locations)\n",
+    "    fraser_polygon = shapely.geometry.polygon.Polygon(f.locations)\n",
+    "    columbia_polygon = shapely.geometry.polygon.Polygon(c.locations)\n",
+    "\n",
+    "    if (peace_polygon.contains(point)):\n",
+    "        return ('Peace')\n",
+    "    \n",
+    "    elif (fraser_polygon.contains(point)):\n",
+    "        return ('Fraser')\n",
+    "        \n",
+    "    elif (columbia_polygon.contains(point)):\n",
+    "        return ('Columbia')\n",
+    "\n",
+    "    else:\n",
+    "        return None\n",
+    "\n",
+    "\n",
+    "def date_widget(descr):\n",
+    "    return DatePicker(description=descr, disabled=False)\n",
+    "\n",
+    "\n",
+    "def build_url(start, end, points):\n",
+    "   \n",
+    "    start_val = '&run_startdate=' + str(start) + '-00'\n",
+    "    end_val = '&stop_date=' + str(end)\n",
+    "    \n",
+    "    lon_val = '&lons=' + ','.join([point.split(', ')[1] for point in points])\n",
+    "    lat_val = '&lats=' + ','.join([point.split(', ')[0] for point in points])\n",
+    "\n",
+    "    url = 'case_id=sample' + start_val + end_val + lon_val + lat_val\n",
```

Comment:
Since we're setting the `case_id` to the name of the watershed, we don't need the `case_id=sample` part.